libs/libc/misc/lib_memfd.c

```diff
@@ -32,20 +33,30 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-#define LIBC_MEM_FD_VFS_PATH \
-          CONFIG_LIBC_TMPDIR "/" CONFIG_LIBC_MEM_FD_VFS_PATH "/%s"
+#ifdef CONFIG_LIBC_MEMFD_ON_TMPFS
+#  define LIBC_MEM_FD_VFS_PATH CONFIG_LIBC_TMPDIR "/" CONFIG_LIBC_MEM_FD_VFS_PATH
+#else
+#  define LIBC_MEM_FD_VFS_PATH CONFIG_LIBC_MEM_FD_VFS_PATH
+#endif
+
+#define LIBC_MEM_FD_VFS_PATH_FMT LIBC_MEM_FD_VFS_PATH "/%s"
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
 int memfd_create(FAR const char *name, unsigned int flags)
 {
-#ifdef CONFIG_FS_TMPFS
+#if defined(CONFIG_LIBC_MEMFD_ON_SHMFS) || (CONFIG_LIBC_MEMFD_ON_TMPFS)
```

Comment:
```suggestion
#if defined(CONFIG_LIBC_MEMFD_ON_SHMFS) || defined(CONFIG_LIBC_MEMFD_ON_TMPFS)
```

Reply:
I change the implementation, please review again, thanks.